core/chains/evm/logpoller/log_poller.go

```diff
@@ -994,20 +1020,40 @@ func (lp *logPoller) getCurrentBlockMaybeHandleReorg(ctx context.Context, curren
 // currentBlockNumber is the block from where new logs are to be polled & saved. Under normal
 // conditions this would be equal to lastProcessed.BlockNumber + 1.
 func (lp *logPoller) PollAndSaveLogs(ctx context.Context, currentBlockNumber int64) {
+	err := lp.pollAndSaveLogs(ctx, currentBlockNumber)
+	if errors.Is(err, commontypes.ErrFinalityViolated) {
+		lp.lggr.Criticalw("Failed to poll and save logs due to finality violation, retrying later", "err", err)
+		lp.finalityViolated.Store(true)
+		lp.SvcErrBuffer.Append(err)
+		return
+	}
+
+	if err != nil {
+		lp.lggr.Errorw("Failed to poll and save logs, retrying later", "err", err)
```

Comment:
food for thought, should we track these errors using a more "observable" approach? For instance, by increasing a prometheus counter whenever pollAndSave or backup fails?

Reply:
I'm not sure that it's really helpful to track this as a metric. Failures of pollAndSave and backup are normal (RPC failed for one of the requests), so we won't be using it on overview dashboards.  We might want to introduce something of a higher level. Like the latest block successfully processed by LogPoller and compare it to one observed by HT or another component. High delta of these two values signals that PollAndSaved failed too often.
In any case it seems to be out of scope for this PR.

Reply:
Yeah, I think the "failed to poll and save logs due to finality violation" one is worth tracking, but as I understand it we're already tracking that indirectly because `Healthy()` will start returning false as soon as it happens. The other ones are fairly normal, will happen due to any sort of temporary network instability which presumably is also tracked elsewhere